ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,172 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseUI:
+    """
+    Base Class for UI Tests
+
+    """
+
+    def __init__(self, driver):
+        self.driver = driver
+        self.screenshots_folder = tempfile.mkdtemp()
+        logger.info(f"screenshots pictures:{self.screenshots_folder}")
+        self.cnt_screenshot = 0
+
+    def do_click(self, by_locator, type=By.XPATH, timeout=30, screenshot=True):
+        """
+        Click on Button/link on OpenShift Console
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        type(By): Set of supported locator strategies.
+        timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+        screenshot(bool): True- take screenshot, False-no take screenshot
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.click()
+        if screenshot:
+            self.take_screenshot()
+
+    def do_send_keys(self, by_locator, text, type=By.XPATH, timeout=30):
+        """
+        Send text to element on OpenShift Console
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        text(str): Send text to element
+        type(By): Set of supported locator strategies
+        timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.send_keys(text)
+
+    def is_expanded(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Check whether an element is in an expanded or collapsed state
+
+        Args:
+            by_locator(str): Command that tells Selenium IDE which
+            GUI element needs to operate on.
+            type(By): Set of supported locator strategies
+            timeout(int): Looks for a web element repeatedly until timeout(sec) happens.
+
+        return:
+            (str): "true"-element expanded, "false"-element collapsed
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return element.get_attribute("aria-expanded")
+
+    def choose_expanded_mode(self, mode, by_locator, type):
+        """
+        Select the element mode (expanded or collapsed)
+
+        by_locator(str): Command that tells Selenium IDE which
+        GUI element needs to operate on.
+        type(By): Set of supported locator strategies
+        mode(str): Element mode (expanded or collapsed)
+
+        """
+        current_mode = self.is_expanded(by_locator=by_locator, type=type)
+        if mode != current_mode:
+            self.do_click(by_locator=by_locator, type=type)
+
+    def take_screenshot(self):
+        """
+        Take screenshot using python code
```

Comment:
@petr-balogh @dahorak 
Where to copy the SCREENSHOTS? 
Can I copy a link to the jenkins log?

Reply:
Did I understand it correctly, that it takes screenshot after every click? How many screenshots it will be (how much space it will take the screenshots from one test run)?
I think, it will be good to be able to enable/disable the snapshots creation from config file.
Also it might be helpful for debugging to take screenshot in case of failure (if it is possible to implement it this way).

And yes, it will be worth to print info about created screenshot to log (including file path) - not sure if as debug or info level?

Otherwise I think, that saving them to the `log_dir` is good place.

Reply:
@dahorak 
I added a flag of screenshot (True/False) on `do_click` method.

Reply:
I meant to make it configurable from config file - to be able to change it for each test execution.

Reply:
here?
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/conf/default_config.yaml